scripts/ci/build_emscripten.sh

```diff
@@ -61,12 +59,20 @@ then
 	echo -n "$CIRCLE_SHA1" >commit_hash.txt
 fi
 
+# Disable warnings for unqualified "move()" calls, introduced and enabled by
+# default in clang-16 which is what the emscripten docker image uses.
+# Additionally, disable the warning for unknown warnings here, as this script is
+# also used with earlier clang versions.
+CMAKE_CXX_FLAGS="-Wno-unqualified-std-cast-call -Wno-unknown-warning-option"
```

Comment:
I am not super happy about the `-Wno-unknown-warnings-option`-option as it might hide problems we don't see otherwise.

However, as it is only enabled for this particular script, it seemed okay to me.
Alternative approaches:

* Check if the current clang used supports the flag and only then use (and disable) it
  * Requires an extra file (empty `main()`) or other method to do that check
* Update all our `move()` calls to `std::move()` and if possible add this or a similar option in our normal compile-pipeline so that we don't accidentally add a `move()` without std
  * Not sure if such a flag exists before clang-16 and for gcc at all. If it doesn't we risk breaking b_ems .. which the CI would notice though...
* Update the image used by b_ems to use the latest clang as this seems to be the only other automated use of this script

I don't really know where the image for b_ems comes from (in the config it just says `- image: << pipeline.parameters.emscripten-docker-image >>`) so my last two points might not apply, depending how all those images are setup...

Reply:
> It's built from scripts/docker/buildpack-deps/Dockerfile.emscripten, which currently extends emscripten/emsdk:2.0.33.

Which this PR modifies and updates... the whole reason for that flag is because this image does use the latest clang..
I only noticed that the b_ems failed earlier so I assumed it must be a different image as the buildpack stuff worked.
I suppose it simply didn't update the image yet at that point.
If that's the case we can just remove the `-Wno-unknown-warning-option` after the image is updated.

> I guess we could add a clang update step to the dockerfile.

Not required at all, the ems people use the master branch from clang in their image ;)

Reply:
See the last comment at https://github.com/emscripten-core/emscripten/pull/16386#issuecomment-1224298275

Reply:
> I only noticed that the b_ems failed earlier so I assumed it must be a different image as the buildpack stuff worked.
> I suppose it simply didn't update the image yet at that point.

Yeah. We still have the old hash in `config.yml`. But thanks for pointing that out. I would completely forget that we also need to update the hash the in a separate PR - can you create such a PR?

Reply:
> If that's the case we can just remove the `-Wno-unknown-warning-option` after the image is updated.

So can we remove it?

Reply:
Actually, since after update we'll get latest clang, does that mean we can remove both?

If not, there should be a TODO comment saying when we'll be able to remove the hack.

Reply:
Technically yes, but only after this PR updated the image.
I will just create another PR that will remove it right after this is merged

Reply:
Wait, both? Why both? `unqualified-std-cast-call` is enabled by default in the latest clang, that's where the whole issue came from

Reply:
Ah, I assumed that we might not need it because emscripten itself adds `-Wno-unqualified-std-cast-call` (https://github.com/emscripten-core/emscripten/pull/16386). Just wasn't sure. So that applies only to their own code and we do need it too after all?

Reply:
Correct :)